### PR TITLE
Feat: pure proxy list with params

### PR DESCRIPTION
**Pull request by Asmadek**
* use json for chains with proxy and pure proxy support instead of arrays

### CODE REVIEW

File: scripts/buildChainsJson.js [hunk 1]
```diff
@@ -154,6 +139,20 @@ function getTransformedData(rawData) {
         return explorer;
       });
 
+      const proxyData = PROXY_LIST.find(p => p.chainId === chain.chainId);
+
+      if (proxyData) {
+        options.concat(proxyData.options)
```
Comment (tuul-wq) [inline scripts/buildChainsJson.js]: `concat` returns a new array, we need `push(...proxyData.options)` 
[doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat)

File: scripts/buildChainsJson.js [hunk 2]
```diff
@@ -154,6 +139,20 @@ function getTransformedData(rawData) {
...
+      }
+
+      if (proxyData?.externalApi) {
+        externalApi = {
```
Comment (tuul-wq) [inline scripts/buildChainsJson.js:149]: `Object.assign(externalApi, proxyData.externalApi)` looks a bit cleaner

**Review by tuul-wq — COMMENTED** [review]
Few suggestions

File: scripts/buildChainsJson.js (hunk 2 above)
Comment (Asmadek) [reply scripts/buildChainsJson.js:149]: `externalApi` maybe null, and it throw an error

File: scripts/buildChainsJson.js [hunk 3]
```diff
@@ -118,8 +111,8 @@ function getTransformedData(rawData) {
   });
 
   return filteredData.map(chain => {
-      const externalApi = filterObjectByKeys(chain.externalApi, ['staking', 'history']);
-      const options = [];
+      let externalApi = filterObjectByKeys(chain.externalApi, ['staking', 'history']);
+      let options = [];
```
Comment (tuul-wq) [inline scripts/buildChainsJson.js:115]: I don't see `options` reference being changed in the following code, `const` is still valid

File: scripts/buildChainsJson.js [hunk 4]
```diff
@@ -154,6 +139,20 @@ function getTransformedData(rawData) {
...
+      const proxyData = PROXY_LIST.find(p => p.chainId.includes(chain.chainId));
+
+      if (proxyData) {
+        options.push(...proxyData.options)
+      }
+
+      if (proxyData?.externalApi) {
+        externalApi = {
+          ...externalApi,
+          ...proxyData.externalApi
```
Comment (tuul-wq) [inline scripts/buildChainsJson.js:151]:
```suggestion
        externalApi = { ...externalApi, ...proxyData.externalApi };
```

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**